Dalamud/Game/Network/Internal/NetworkHandlers.cs

```diff
@@ -98,81 +226,67 @@ protected void Dispose(bool shouldDispose)
 
-    private IObservable<MarketTaxRates> OnMarketTaxRates()
-    {
-        return this.OnNetworkMessage()
-                   .Where(message => message.Direction == NetworkMessageDirection.ZoneDown)
-                   .Where(message => message.Opcode == message.DataManager?.ServerOpCodes["MarketTaxRates"])
-                   .Where(message =>
-                   {
-                       // Only some categories of the result dialog packet contain market tax rates
-                       var category = (uint)Marshal.ReadInt32(message.Data);
-                       return category == 720905;
-                   })
-                   .Select(message => MarketTaxRates.Read(message.Data))
-                   .Where(taxes => taxes.Category == 0xb0009);
-    }
+        if (this.configuration.DutyFinderTaskbarFlash)
+            Util.FlashWindow();
 
-    private IObservable<MarketBoardPurchaseHandler> OnMarketBoardPurchaseHandler()
-    {
-        return this.OnNetworkMessage()
-                   .Where(message => message.Direction == NetworkMessageDirection.ZoneUp)
-                   .Where(message => message.Opcode == message.DataManager?.ClientOpCodes["MarketBoardPurchaseHandler"])
-                   .Select(message => MarketBoardPurchaseHandler.Read(message.Data));
-    }
+        var cfConditionSheet = Service<DataManager>.Get().GetExcelSheet<ContentFinderCondition>()!;
+        var cfCondition = cfConditionSheet.GetRow(conditionId);
 
-    private IObservable<MarketBoardPurchase> OnMarketBoardPurchase()
-    {
-        return this.OnNetworkMessage()
-                   .Where(message => message.Direction == NetworkMessageDirection.ZoneDown)
-                   .Where(message => message.Opcode == message.DataManager?.ServerOpCodes["MarketBoardPurchase"])
-                   .Select(message => MarketBoardPurchase.Read(message.Data));
-    }
+        if (cfCondition == null)
+        {
+            Log.Error("CFC key {ConditionId} not in Lumina data", conditionId);
+            goto ORIGINAL;
+        }
 
-    private IObservable<NetworkMessage> OnCfNotifyPop()
-    {
-        return this.OnNetworkMessage()
-                   .Where(message => message.Direction == NetworkMessageDirection.ZoneDown)
-                   .Where(message => message.Opcode == message.DataManager?.ServerOpCodes["CfNotifyPop"]);
+        var cfcName = cfCondition.Name.ToString();
+        if (cfcName.IsNullOrEmpty())
+        {
+            cfcName = "Duty Roulette";
+            cfCondition.Image = 112324;
+        }
+
+        Task.Run(() =>
+        {
+            if (this.configuration.DutyFinderChatMessage)
+            {
+                Service<ChatGui>.GetNullable()?.Print($"Duty pop: {cfcName}");
+            }
+
+            this.CfPop.InvokeSafely(cfCondition);
+        }).ContinueWith(
+            task => Log.Error(task.Exception, "CfPop.Invoke failed"),
+            TaskContinuationOptions.OnlyOnFaulted);
+
+        ORIGINAL:
```

Comment:
goto considered harmful, please use try/finally

Reply:
Could wrap it in an else bracket